romeo/src/state.rs

```diff
@@ -444,11 +440,18 @@ impl State {
 
 		withdrawals
 			.iter_mut()
-			.filter_map(|withdrawal| match withdrawal.fulfillment.as_mut() {
-				None => {
-					withdrawal.fulfillment = Some(TransactionRequest::Created);
-					Some(Task::CreateFulfillment(withdrawal.info.clone()))
-				}
+			.filter_map(|withdrawal| match withdrawal.burn {
+				Some(TransactionRequest::Acknowledged {
+					status: TransactionStatus::Confirmed,
+					..
+				}) => match withdrawal.fulfillment.as_mut() {
+					None => {
+						withdrawal.fulfillment =
+							Some(TransactionRequest::Created);
+						Some(Task::CreateFulfillment(withdrawal.info.clone()))
```

Comment:
Only create a fulfillment task for requests where burn tx on stacks was successful. Otherwise, do not create a task.